torch/csrc/jit/interpreter.cpp

```diff
@@ -58,12 +58,79 @@ namespace torch { namespace jit {
 void desugarTripCounts(Block * b) {
   for(auto n : b->nodes()) {
     if(n->kind() == kLoop) {
-
-      // remove the trip count from Loop inputs, we don't support it yet
+      // First, replace the max trip count input with the initial trip count.
```

Comment:
Hmm I don't think that what we have now belongs in the interpreter. It emits instructions in the graph, and that's the job of the compiler. This function was supposed to be a simple way of applying a tiny hack to the IR loops to work with the code we have in the interpreter (and assert that we don't use the trip counter).

Reply:
I think having this here makes sense because we want to keep the specification of loop nodes 1:1 with the ONNX: https://github.com/onnx/onnx/blob/master/docs/Operators.md#Loop. This is an implementation detail of the interpreter, and represents a delta from the "standardized" schema. The fact that we have this probably indicates that we need to change the standardized schema so that it matches, but I think that is something for the future